### PR TITLE
Release 3.3.4

### PR DESCRIPTION
**Pull request by hardbyte**
Would appreciate a few people testing out the beta before we release the (hopefully) final 3.x release.

Happy for any last features or fixes to make it in too.

```
pip install python-can==3.3.4b0
```

### CODE REVIEW

**Comment by tamenol** [comment]
I see nothing wrong in my application when using the latest beta version. I'm using the PCAN interface.

File: CHANGELOG.txt [hunk 1]
```diff
@@ -1,3 +1,15 @@
+Version 3.3.4
+====
+
+Last call for Python2 support.
+
+* #850 Fix socket.error is a deprecated alias of OSError used on Python versions lower than 3.3.
+* #846 Use inter-process mutex to prevent concurrent neoVI device open.
```
Comment (pierreluctg) [inline CHANGELOG.txt:7]: I see this in the change log here but https://github.com/hardbyte/python-can/pull/621 is not included in this release branch

Comment (hardbyte) [reply CHANGELOG.txt:7]: Ahh my mistake. I'll remove the line from the changelog as I closed that issue without backporting.

Comment (pierreluctg) [reply CHANGELOG.txt]: Can we include the change in 3.3.4?

Comment (hardbyte) [reply CHANGELOG.txt]: Sure I'm happy to merge it in if you want to open a PR against this `release-3.3.4` branch

Comment (pierreluctg) [reply CHANGELOG.txt]: Please see https://github.com/hardbyte/python-can/pull/903

**Comment by hardbyte** [comment]
This will automatically publish the release if/when someone else approves the PR.

**Comment by felixdivo** [comment]
I currently do not have any CAN hardware at all in my home office. So I can only say that the changes *look* fine, not that they actually *work*.

File: can/bus.py [hunk 2]
```diff
@@ -247,13 +247,21 @@ def _send_periodic_internal(self, msg, period, duration=None):
...
             Stop tracking the stopped tasks.
         """
         for task in self._periodic_tasks:
-            task.stop(remove_task=remove_tasks)
+            # we cannot let `task.stop()` modify `self._periodic_tasks` while we are
+            # iterating over it (#634)
+            task.stop(remove_task=False)
+
+        if remove_tasks:
+            self._periodic_tasks = []
```
Comment (zariiii9003) [inline can/bus.py:264]: I would prefer `self._periodic_tasks.clear()` here

File: setup.py [hunk 3]
```diff
@@ -7,6 +7,7 @@
 
 from __future__ import absolute_import
 
+import platform
```
Comment (zariiii9003) [inline setup.py:10]: `platform` is imported but not used

**Review by zariiii9003 — CHANGES_REQUESTED** [review]
I would remove the `platform` import in `setup.py` (unless i missed something). Otherwise this looks good.

File: can/bus.py (hunk 2 above)
Comment (hardbyte) [reply can/bus.py:264]: Me too, but this release still has to support Python 2

**Review by zariiii9003 — APPROVED** [review]

**Pull request merged**